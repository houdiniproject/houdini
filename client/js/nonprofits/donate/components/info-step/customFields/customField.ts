// License: LGPL-3.0-or-later
import { CustomFieldDescription } from "../../../types";
const h = require('snabbdom/h');


const typeToFormInputName = {
  'supporter': 'customFields'
}

export default function customField(field: CustomFieldDescription) : ReturnType<typeof h> {
  return h('input', {
    props: {
      name: `${typeToFormInputName[field.type]}[${field.name}]`,
      placeholder: field.label
    }
  });
}