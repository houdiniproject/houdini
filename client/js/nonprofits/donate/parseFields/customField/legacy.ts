// License: LGPL-3.0-or-later
import {map as Rmap, split as Rsplit, trim as Rtrim} from 'ramda';
import { CustomFieldDescription } from '.';

export function parseCustomField(f:string) :CustomFieldDescription {
  const [name, label] = Rmap(Rtrim, Rsplit(':', f))
  return {name, label: label || name, type: 'supporter'};
}