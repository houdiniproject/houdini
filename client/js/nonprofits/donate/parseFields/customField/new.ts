// License: LGPL-3.0-or-later
import { CustomFieldDescription } from ".";

export function parseCustomField(input:string) : CustomFieldDescription {
  const [name, ...rest] = input.split(":").map((s) => s.trim())
  const label = rest.length > 0 ? rest[0] : null;

  return {name, label: label || name, type: 'supporter' } as CustomFieldDescription;
};
