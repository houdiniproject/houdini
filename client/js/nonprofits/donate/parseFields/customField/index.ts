// License: LGPL-3.0-or-later
import { parseCustomField } from "./legacy";
import has from 'lodash/has';
import get from 'lodash/get';

export interface CustomFieldDescription {
  name: NonNullable<string>;
  label: NonNullable<string>;
  type: 'supporter';
}

export function isCustomFieldDescription(item:unknown) : item is CustomFieldDescription {
  return typeof item == 'object' && 
    has(item, 'name') && 
    has(item, 'label') && 
    ['supporter'].includes(get(item, 'type'));
}


export default parseCustomField;