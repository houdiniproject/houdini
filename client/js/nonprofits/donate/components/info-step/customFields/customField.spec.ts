// License: LGPL-3.0-or-later
import customField from "./customField"

describe('customField', () => { 
  it('sets the the field correctly', () => {
    expect(customField({name: "Name of Field", label: "Field Label", type: 'supporter'})).toMatchSnapshot();
  })
});