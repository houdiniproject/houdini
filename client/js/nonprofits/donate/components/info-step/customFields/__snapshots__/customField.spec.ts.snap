// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customField sets the the field correctly 1`] = `
{
  "children": undefined,
  "data": {
    "props": {
      "name": "customFields[Name of Field]",
      "placeholder": "Field Label",
    },
  },
  "elm": undefined,
  "key": undefined,
  "sel": "input",
  "text": undefined,
}
`;
